sdk/monitor/monitor-opentelemetry/src/metrics/quickpulse/utils.ts

```diff
@@ -219,80 +253,255 @@ function getIso8601Duration(milliseconds: number): string {
   const grpcStatusCode = span.attributes[SEMATTRS_RPC_GRPC_STATUS_CODE];
-  let url = "";
-  let code = "";
-  if (span.kind === SpanKind.SERVER || span.kind === SpanKind.CONSUMER) {
-    if (httpMethod) {
-      url = getUrl(span.attributes);
-      const httpStatusCode = span.attributes[SEMATTRS_HTTP_STATUS_CODE];
-      if (httpStatusCode) {
-        code = String(httpStatusCode);
-      }
-    } else if (grpcStatusCode) {
-      code = String(grpcStatusCode);
+  if (httpMethod) {
+    requestData.Url = getUrl(span.attributes);
+    const urlObj = new URL(requestData.Url);
+    requestData.Name = `${httpMethod} ${urlObj.pathname}`;
+    const httpStatusCode = span.attributes[SEMATTRS_HTTP_STATUS_CODE];
+    if (httpStatusCode) {
+      requestData.ResponseCode = Number(httpStatusCode);
     }
+  } else if (grpcStatusCode) {
+    requestData.ResponseCode = Number(grpcStatusCode);
+  }
+  requestData.Success = span.status.code !== SpanStatusCode.ERROR && requestData.ResponseCode < 400;
+  return requestData;
+}
 
-    document = {
-      documentType: KnownDocumentType.Request,
-      name: span.name,
-      url: url,
-      responseCode: code,
-      duration: getIso8601Duration(hrTimeToMilliseconds(span.duration)),
-    };
-  } else {
-    url = getUrl(span.attributes);
+// A slightly modified version of createDependencyData from spanUtils in exporter
+function getDependencyData(span: ReadableSpan): DependencyData {
+  const dependencyData: DependencyData = {
+    Target: "",
+    Duration: hrTimeToMilliseconds(span.duration),
+    Success: span.status.code !== SpanStatusCode.ERROR,
+    Name: span.name,
+    ResultCode: 0,
+    Type: "",
+    Data: "",
+    CustomDimensions: createCustomDimsFromAttributes(span.attributes),
+  };
+
+  if (span.kind === SpanKind.PRODUCER) {
+    dependencyData.Type = DependencyTypes.QueueMessage;
+  }
+  if (span.kind === SpanKind.INTERNAL && span.parentSpanId) {
+    dependencyData.Type = DependencyTypes.InProc;
+  }
+
+  const httpMethod = span.attributes[SEMATTRS_HTTP_METHOD];
+  const dbSystem = span.attributes[SEMATTRS_DB_SYSTEM];
+  const rpcSystem = span.attributes[SEMATTRS_RPC_SYSTEM];
+  // HTTP Dependency
+  if (httpMethod) {
+    const httpUrl = span.attributes[SEMATTRS_HTTP_URL];
+    if (httpUrl) {
+      try {
+        const dependencyUrl = new URL(String(httpUrl));
+        dependencyData.Name = `${httpMethod} ${dependencyUrl.pathname}`;
+      } catch (ex: any) {
+        /* no-op */
+      }
+    }
+    dependencyData.Type = DependencyTypes.Http;
+    dependencyData.Data = getUrl(span.attributes);
     const httpStatusCode = span.attributes[SEMATTRS_HTTP_STATUS_CODE];
     if (httpStatusCode) {
-      code = String(httpStatusCode);
+      dependencyData.ResultCode = Number(httpStatusCode);
+    }
+    let target = getDependencyTarget(span.attributes);
+    if (target) {
+      try {
+        // Remove default port
+        const portRegex = new RegExp(/(https?)(:\/\/.*)(:\d+)(\S*)/);
```

Comment:
You should be able to use [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) to parse URL values instead of doing your own regex validation

Reply:
I copied this part directly from spanUtils.ts in the exporter - should it be changed there too?

Reply:
No need to update, I can see target is not necessary a URL and that is why the Regex is in place, as Otel peer service is one of the possible outcomes